### PR TITLE
Test MCG data reduction

### PR DESCRIPTION
**Pull request by MeridianExplorer**
Test MCG data reduction

### CODE REVIEW

File: ocs_ci/ocs/resources/mcg.py [hunk 1]
```diff
@@ -17,7 +19,11 @@ class MCG(object):
     Wrapper class for the Multi Cloud Gateway's S3 service
     """
 
-    s3_resource, ocp_resource, endpoint, region, access_key_id, access_key, namespace = (None,) * 7
+    (
```
Comment (ebenahar) [inline ocs_ci/ocs/resources/mcg.py]: Shouldn't we do this initialization inside an __init__() func?

File: ocs_ci/ocs/resources/mcg.py [hunk 2]
```diff
@@ -144,3 +161,76 @@ def cli_verify_bucket_exists(self, bucketname):
 
         """
         return bucketname in self.cli_list_all_bucket_names()
+
+    def check_data_reduction(self, bucketname):
+        """
+        Checks whether the data reduction on the MCG server works properly
```
Comment (ebenahar) [inline ocs_ci/ocs/resources/mcg.py]: please add an empty line blow

File: tests/manage/mcg/test_write_to_bucket.py [hunk 3]
```diff
@@ -51,3 +52,27 @@ def test_write_file_to_bucket(self, mcg_obj, awscli_pod, bucket_factory, uploade
...
+        """
+        Test data reduction mechanics
+
+        """
+        # TODO: Privatize test bucket
+        download_dir = '/aws/downloaded'
+        synccmd = (
+            f"aws s3 sync s3://{constants.TEST_FILES_BUCKET} {download_dir} --no-sign-request"
+        )
+        assert 'download' in awscli_pod.exec_cmd_on_pod(command=synccmd, out_yaml_format=False)
```
Comment (ebenahar) [inline tests/manage/mcg/test_write_to_bucket.py]: Please add an assertion message

File: tests/manage/mcg/test_write_to_bucket.py [hunk 4]
```diff
@@ -51,3 +52,27 @@ def test_write_file_to_bucket(self, mcg_obj, awscli_pod, bucket_factory, uploade
...
+        for bucket in bucket_factory(5):
+            synccmd = f'sync {download_dir} s3://{bucket.name}'
+            awscli_pod.exec_cmd_on_pod(
+                command=craft_s3_command(mcg_obj, synccmd),
+                out_yaml_format=False,
+                secrets=[mcg_obj.access_key_id, mcg_obj.access_key, mcg_obj.s3_endpoint]
+            )
+            bucketname = bucket.name
+
+        assert mcg_obj.check_data_reduction(bucketname)
```
Comment (ebenahar) [inline tests/manage/mcg/test_write_to_bucket.py]: same

File: ocs_ci/ocs/resources/mcg.py (hunk 1 above)
Comment (MeridianExplorer) [reply ocs_ci/ocs/resources/mcg.py]: Strictly speaking, there's no need at all to declare those valuables in Python - it's mostly for documentation and readability. We can take this off completely - I tend to prefer leaving it out.

Comment (ebenahar) [reply ocs_ci/ocs/resources/mcg.py]: Apart from readability, I am too not aware of any other reason. I'm ok with both

File: ocs_ci/ocs/resources/mcg.py [hunk 5]
```diff
@@ -28,28 +34,47 @@ def __init__(self):
...
+            creds_secret_obj.get('data').get('password')
+        ).decode('utf-8')
+
+        self.noobaa_token = self.send_rpc_query(
+            'auth_api', 'create_auth', params={
+                'role': 'admin',
+                'system': 'noobaa',
+                'email': self.noobaa_user,
+                'password': self.noobaa_password
+            }).json().get('reply').get('token')
```
Comment (RazTamir) [inline ocs_ci/ocs/resources/mcg.py]: ).json().get('reply').get('token') 

in new line

File: ocs_ci/ocs/resources/mcg.py [hunk 6]
```diff
@@ -168,3 +193,81 @@ def cli_verify_bucket_exists(self, bucketname):
...
+            }
+            request_str = json.dumps(payload)
+            resp = requests.post(url=self.mgmt_endpoint, data=request_str, verify=False)
+            bucket_data_reduced = resp.json().get('reply').get('data').get('size_reduced')
+
+            logger.info(
+                'Overall bytes stored: ' + str(bucket_data) + '. Amount reduced: ' + str(bucket_data_reduced)
+            )
+
+            return bucket_data, bucket_data_reduced
```
Comment (RazTamir) [inline ocs_ci/ocs/resources/mcg.py]: this function should return bool

Comment (MeridianExplorer) [reply ocs_ci/ocs/resources/mcg.py]: Woah, that line should've been deleted. Fixed. Thank you!

Comment (MeridianExplorer) [reply ocs_ci/ocs/resources/mcg.py]: Please note this is the inside function, `_retrieve_data_reduction`. It retrieves the data reduction data - `check_data_reduction` returns bool

Comment (RazTamir) [reply ocs_ci/ocs/resources/mcg.py]: ack

**Pull request merged**